core/lib/spree/core/product_duplicator.rb

```diff
@@ -70,5 +70,12 @@ def reset_properties
         end
       end
     end
+
+    def generate_new_sku(sku)
+      loop do
+        sku = "COPY OF #{sku}"
+        break sku unless Spree::Variant.where(sku: sku).first
```

Comment:
How do you get of this loop?

It is better to add a unit test in the Product Duplicatior than a feature test. Feature tests are expensive.